commit 7bc3807fe1d0694caf59dec983ac5809441cc9ca
Author: George Shen <george.shen@amd.com>
Date:   Tue Feb 18 19:15:55 2020 -0500

    drm/amd/display: Workaround to do HDCP authentication twice on certain displays
    
    [Why]
    When transitioning from SST to MST, the HDCP repeater in some MST
    displays will enter a bad state. The HDCP repeater is recovered after
    failing and performing authentication again.
    
    [How]
    Add monitor patch to trigger HDCP authentication failure after
    encryption is enabled and re-authenticate.
    
    Signed-off-by: George Shen <george.shen@amd.com>
    Reviewed-by: Wenjing Liu <Wenjing.Liu@amd.com>
    Acked-by: Rodrigo Siqueira <Rodrigo.Siqueira@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>

diff --git a/drivers/gpu/drm/amd/display/modules/inc/mod_hdcp.h b/drivers/gpu/drm/amd/display/modules/inc/mod_hdcp.h
index 891bca555e17..bb855ea5d5a3 100644
--- a/drivers/gpu/drm/amd/display/modules/inc/mod_hdcp.h
+++ b/drivers/gpu/drm/amd/display/modules/inc/mod_hdcp.h
@@ -157,7 +157,8 @@ struct mod_hdcp_display_adjustment {
 struct mod_hdcp_link_adjustment_hdcp1 {
 	uint8_t disable			: 1;
 	uint8_t postpone_encryption	: 1;
-	uint8_t reserved		: 6;
+	uint8_t min_auth_retries_wa : 1;
+	uint8_t reserved		: 5;
 };
 
 enum mod_hdcp_force_hdcp_type {